### PR TITLE
lxd: Prevent internal cluster migration of instances with backups

### LINKED ISSUE #7959: Prevent internal cluster migration of instances with backups
Backups cannot be moved, so if an instance has an existing backup, we need to have the user delete those backups prior to moving the instance.

### PR DESCRIPTION
**Pull request by DDEFISHER**
Fixes #7959

### CODE REVIEW

**Comment by DDEFISHER** [comment]
Open to ideas about testing. `test/suites/clustering.sh` seems like it could work but AFAICT `lxc export` deletes the backup after the tar.gz is generated.

**Comment by stgraber** [comment]
You can use `lxc query -X POST /1.0/instances/NAME/backups` to create a non-expiring backup, that should work.

Also, your commit message needs the `Signed-off-by: Firstname Lastname <email>` line.
Doing `git commit --amend -s` followed by `git push --force` should do the trick.

**Review by tomponline — APPROVED** [review]

**Comment by DDEFISHER** [comment]
I was having trouble getting these tests to run in my environment but hoping they will pass here.

**Comment by stgraber** [comment]
jenkins: test this please

File: test/suites/clustering.sh [hunk 1]
```diff
@@ -280,6 +280,11 @@ test_clustering_containers() {
   apply_template2=$(LXD_DIR="${LXD_TWO_DIR}" lxc config get egg volatile.apply_template)
   [ "${apply_template1}" =  "${apply_template2}" ] || false
 
+  # Create backup and attempt to move container. Move should fail and container should remain on node3.
+  LXD_DIR="${LXD_THREE_DIR}" lxc query -X POST --wait -d '{\"name\":\"foo\"}' /1.0/instances/egg/backups
+  LXD_DIR="${LXD_THREE_DIR}" lxc move egg --target node1 | grep -q "Instance has backups"
```
Comment (stgraber) [inline test/suites/clustering.sh:285]: Should be change to:
```
! LXD_DIR="${LXD_THREE_DIR}" lxc move egg --target node1 || false
```

So it doesn't trip the shell when exiting non-zero.

Comment (DDEFISHER) [reply test/suites/clustering.sh]: That makes sense. Fixed.

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
@stgraber the `jenkins: test this please` does not seem to trigger  a test suite run on this PR, is it a permissions issue? Thanks

**Comment by stgraber** [comment]
Possibly, I'll have to check how that stuff is configured in Jenkins again...

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
Looks good, thanks!

**Pull request merged**